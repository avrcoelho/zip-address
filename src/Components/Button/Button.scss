@import '../../Scss/root';

.button {
	@media screen and (max-width: $sm-beakpoint) { width: 100% }
	@include rfs(22);
	border: none;
	border-radius: $border-radius;
	cursor: pointer;
	display: inline-block;
	line-height: 0;
	padding: .5em 1em;
	outline: none;

	&:not([disabled]) { @include hover-shadow }
	&[disabled] { background-color: lighten($primary-color, 15%) }
}