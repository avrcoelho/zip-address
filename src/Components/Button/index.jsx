import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

export default function Button({ big, className, children, ...rest }) {
	return (
		<button
			className={`${big ? 'button--big ' : ''}${className ? `${className} ` : '' }button bg-primary color-secondary`}
			{...rest}
		>
			<span className="button__content">
				{children}
			</span>
		</button>
	)
}

Button.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node
	]).isRequired,
	big: PropTypes.bool,
	className: PropTypes.string
}