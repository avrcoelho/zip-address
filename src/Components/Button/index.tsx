import * as React from 'react';
import * as PropTypes from 'prop-types';

import { ButtonProps, ButtonTypes } from '../../../types';

import './Button.scss';

export default function Button({ big, className, children , type, ...rest }:ButtonProps) {
	return (
		<button
			className={`${big ? 'button--big ' : ''}${className ? `${className} ` : '' }button bg-primary color-secondary`}
			type={type ? type : ButtonTypes.button}
			{...rest}
		>
			<span className="button__content">
				{children}
			</span>
		</button>
	)
}

Button.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node,
		PropTypes.string
	]).isRequired,
	big: PropTypes.bool,
	className: PropTypes.string
}