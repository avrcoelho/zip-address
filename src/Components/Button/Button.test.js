import React from 'react';
import 'jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

import Button from './index';

afterEach(cleanup);

describe('<Button />', () => {
	it('should mount properly', () => {
		const { container } = render(
			<Button>
				Test
			</Button>
		);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('shoud mount with big property', () => {
		const { getByTestId } = render(
			<Button big>
				Test
			</Button>
		);

		expect(getByTestId('button')).toHaveClass('button--big');
	});

	it('shoud mount with custom className', () => {
		const { getByTestId } = render(
			<Button className="button--test">
				Test
			</Button>
		);

		expect(getByTestId('button')).toHaveClass('button--test');
	});
});