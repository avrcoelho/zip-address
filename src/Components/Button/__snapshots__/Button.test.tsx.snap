// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should mount properly 1`] = `
<Button>
  <button
    className="button bg-primary color-secondary"
    type="button"
  >
    <span
      className="button__content"
    >
      Test
    </span>
  </button>
</Button>
`;
